send_webhooks.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# workflow-helpers
+# Copyright(C) 2022 Kevin Postlethwait
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""This script sends all webhooks in a directory, each is represented by a JSON object."""
+
+import json
+import logging
+import os
+
+from kubernetes import kubernetes as k8
+
+import requests
+
+dir = os.getenv["WEBHOOK_DIR"]
+document_id = os.getenv["DOCUMENT_ID"]
+secret_namespace = os.getenv["THOTH_BACKEND_NAMESPACE"]
+
+_LOGGER = logging.getLogger("thoth.send_webhooks")
+
+for f_name in os.listdir(dir):
+    with open(os.path.join(dir, f_name), "r") as f:
+        webhook = json.load(f)
+    contents = {
+        "client_data": webhook["client_data"],
+        "document_id": document_id,
+    }
+    headers = {"Authorization": webhook["Authorization"]}
+    try:
+        requests.post(url=webhook["callback_url"], data=contents, headers=headers)
+    except Exception as e:
+        _LOGGER.exception(e)
+
+k8.client.CoreV1Api.delete_namespaced_secret(name=f"callback-{document_id}", namespace=secret_namespace)
```

Comment:
secret would be deleted even if exception occures as we are not raising it.
is this intentional ?

Reply:
I don't think the onus is on us if webhook fails to deliver, the most common reasons it would fail in my opinion is if an invalid URL or authentication is provided. Let me know what you think.